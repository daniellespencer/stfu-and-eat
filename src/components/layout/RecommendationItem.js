import React from "react";

import { Card, Button } from "react-bootstrap";

const RecommendationItem = ({ restaurant }) => {
  const onClick = () => {
    const url = restaurant.website;
    window.open(url, "_blank");
  };

  const onClickDirections = () => {
    const url = `https://www.google.com/maps/dir//${restaurant.address}`;
    window.open(url, "_blank");
  };

  return (
    <div>
      <Card className='shadow mb-5 mt-5 '>
        <Card.Img
          variant='top'
          src={require("../../images/logo.jpg")}
          className='img-fluid'
        />
        <Card.Body>
          <Card.Title>{restaurant.name}</Card.Title>
          <Card.Text>
            Neighborhood: {restaurant.neighborhood} <br />
            Cuisine: {restaurant.cuisine}
          </Card.Text>
          <Button
            variant='primary'
            className='btn-primary btn-sm rec-item-button'
            onClick={onClick}
          >
            Restaurant website
          </Button>
          <Button
            variant='outline-primary'
            className='btn-sm rec-item-button-directions'
            onClick={onClickDirections}
          >
            Directions
          </Button>
        </Card.Body>
      </Card>
    </div>
  );
};

export default RecommendationItem;
