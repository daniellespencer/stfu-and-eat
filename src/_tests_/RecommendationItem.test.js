import React from 'react';
import { render, cleanup, fireEvent } from '@testing-library/react';
import renderer, { act } from 'react-test-renderer';
import { mount } from 'enzyme';
import ReactDOM from 'react-dom';

import RecommendationItem from '../components/layout/RecommendationItem';
export const flushPromises = () => new Promise(setImmediate)
const restaurant = {
    "address": "2901 Salena St, St. Louis, MO 63118",
    "cuisine": "American",
    "id": "5ec1878e67e5bfad7604154f",
    "name": "Benton Park Cafe",
    "neighborhood": "Benton Park",
    "website": "http://www.bentonparkcafe.com/"
  }

describe("RecommendationItem", () => {
    beforeEach(() => {jest.useFakeTimers()})
    afterEach(() => {
        jest.clearAllTimers()
        cleanup()
    });
    // Set up
    
    // execution

    // assertion
    it("renders restaurant recommendation", async () => {
        fetch.mockResponse(JSON.stringify({result: [
         restaurant
        ]}))
        let restaurantRecommendation

        await act( async () => {
            restaurantRecommendation = renderer.create(<RecommendationItem restaurant={restaurant} />)
            await flushPromises()
        });
        

        expect(restaurantRecommendation.toJSON()).toMatchSnapshot()
    });
})    