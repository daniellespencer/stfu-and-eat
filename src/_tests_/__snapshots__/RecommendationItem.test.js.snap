// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecommendationItem renders restaurant recommendation 1`] = `
<div>
  <div
    className="shadow mb-5 mt-10  card"
  >
    <img
      className="card-img-top img-fluid"
      src="logo.jpg"
    />
    <div
      className="card-body"
    >
      <div
        className="card-title h5"
      >
        Benton Park Cafe
      </div>
      <p
        className="card-text"
      >
        Neighborhood: 
        Benton Park
         
        <br />
        Cuisine: 
        American
      </p>
      <button
        className="btn-primary btn-sm rec-item-button btn btn-primary"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        Restaurant website
      </button>
      <button
        className="btn-sm rec-item-button-directions btn btn-outline-primary"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        Directions
      </button>
    </div>
  </div>
</div>
`;
